agent/stats/engine_unix_test.go

```diff
@@ -124,3 +127,83 @@ func TestNetworkModeStatsAWSVPCMode(t *testing.T) {
 		}
 	}
 }
+
+func TestGetContainerStatusMessage(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	oGetStatusMessageReportingPath := getStatusMessageReportingPath
+	defer func() {
+		getStatusMessageReportingPath = oGetStatusMessageReportingPath
+	}()
+
+	testContainer := &apicontainer.DockerContainer{
+		DockerID: "123",
+		Container: &apicontainer.Container{
+			Name: "firelens_v2_container",
+			FirelensConfig: &apicontainer.FirelensConfig{
+				Version: "v2",
+			},
+		},
+	}
+
+	engine := NewDockerStatsEngine(&cfg, nil, nil)
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	engine.ctx = ctx
+	engine.dockerIdToStatusMessageSince = map[string]time.Time{}
+
+	testCases := []struct {
+		name          string
+		statusMessage string
+		isNil         bool
+		modifyFile    bool
+		fileNotExists bool
+	}{
+		{
+			name:          "getContainerStatusMessage returns new message",
+			statusMessage: "new status message",
+			isNil:         false,
+			modifyFile:    true,
+		},
+		{
+			name:          "getContainerStatusMessage returns updated message",
+			statusMessage: "updated status message",
+			isNil:         false,
+			modifyFile:    true,
+		},
+		{
+			name:          "getContainerStatusMessage returns empty string for no update",
+			statusMessage: "",
+			isNil:         true,
+		},
+		{
+			name:          "getContainerStatusMessage returns empty string when status message file does not exist",
+			statusMessage: "",
+			isNil:         false,
+			fileNotExists: true,
+		},
+	}
+
+	tempfile, _ := ioutil.TempFile("", "firelens_v2_container")
+	getStatusMessageReportingPath = func(dataDirOnHost, taskARN, containerName string) (s string, e error) {
```

Comment:
do you actually need to mock this? it seems that you can create a temp dir with `iotuil.TempDir` and put it in `engine.cfg.DataDirOnHost`, and then the test should just put status message file there. also, it seems you can add `defer os.Remove()` right after calling ioutil so that you don't need to call it multiple times later